### PR TITLE
chore: add events to register client and staff

### PR DESCRIPTION
**Pull request by EugeneDevv**

### CODE REVIEW

**Review by johnngugi — APPROVED** [review]

File: lib/application/redux/actions/create_group/create_group_action.dart [hunk 1]
```diff
@@ -61,6 +64,13 @@ class CreateGroupAction extends ReduxAction<AppState> {
       }
 
       this.onSuccess();
+      await AnalyticsService().logEvent(
+        name: addNewGroup,
+        eventType: AnalyticsEventType.INTERACTION,
+        parameters: <String, dynamic>{
+          'input': createGroupPayload.toJson(),
```
Comment (abiudrn) [inline lib/application/redux/actions/create_group/create_group_action.dart:71]: Add these parameters individually. GA will not collect this metric if there is a nested structure in the payload

add them individually like so
```
{
 "group name": "test group",
"client types": ""
}
```

File: lib/application/redux/actions/register_client/register_client_action.dart [hunk 2]
```diff
@@ -66,6 +69,13 @@ class RegisterClientAction extends ReduxAction<AppState> {
       }
 
       this.onSuccess();
+      await AnalyticsService().logEvent(
+        name: registerClient,
+        eventType: AnalyticsEventType.INTERACTION,
+        parameters: <String, dynamic>{
+          'input': registerClientPayload.toJson(),
```
Comment (abiudrn) [inline lib/application/redux/actions/register_client/register_client_action.dart:76]: Same here

File: lib/application/redux/actions/register_staff/register_staff_action.dart [hunk 3]
```diff
@@ -68,6 +71,13 @@ class RegisterStaffAction extends ReduxAction<AppState> {
       }
 
       this.onSuccess();
+      await AnalyticsService().logEvent(
+        name: registerStaff,
+        eventType: AnalyticsEventType.INTERACTION,
+        parameters: <String, dynamic>{
+          'input': registerStaffPayload.toJson(),
```
Comment (abiudrn) [inline lib/application/redux/actions/register_staff/register_staff_action.dart:78]: Same here

**Review by abiudrn — CHANGES_REQUESTED** [review]

File: lib/application/redux/actions/create_group/create_group_action.dart (hunk 1 above)
Comment (EugeneDevv) [reply lib/application/redux/actions/create_group/create_group_action.dart:71]: Noted

**Review by johnngugi — APPROVED** [review]

**Review by abiudrn — APPROVED** [review]

**Pull request merged**